src/layers/ConstructionLayer/ConstructionClusterLayer.js

```diff
@@ -0,0 +1,80 @@
+import ClusterSource from 'ol/source/Cluster';
+import Style from 'ol/style/Style';
+import Text from 'ol/style/Text';
+import Fill from 'ol/style/Fill';
+import Circle from 'ol/style/Circle';
+import GeoJSON from 'ol/format/GeoJSON';
+import ConstructionLayer from './ConstructionLayer';
+
+class ConstructionClusterLayer extends ConstructionLayer {
+  constructor(options = {}) {
+    super({
+      ...options,
+    });
+
+    this.onChangeVisible = this.onChangeVisible.bind(this);
+
+    this.olLayer.setSource(
+      new ClusterSource({
+        distance: 90,
+        source: this.olLayer.getSource(),
+        geometryFunction: this.isFeatureVisible,
+      }),
+    );
+  }
+
+  addFeatures(data) {
+    const format = new GeoJSON();
+    const features = format.readFeatures(data);
+    this.olLayer
+      .getSource()
+      .getSource()
+      .clear();
+    this.olLayer
+      .getSource()
+      .getSource()
+      .addFeatures(features);
+  }
```

Comment:
Shouldn't the addFeatures function goes in the `ConstructionLayer` ?  since we use it in the smae source.

Reply:
These two different sources:
`this.olLayer.getSource()` for the single layer and `this.olLayer.getSource().getSource()` for the cluster layer as the cluster source is added as source to the vector layer:

```javascript
this.olLayer.setSource(
  new ClusterSource({
    distance: 90,
    source: this.olLayer.getSource(),
    geometryFunction: this.geometryFunction,
  }),
);
```
I will simplify it.